### PR TITLE
#24 locale to localStorage

### LINKED ISSUE #24: Persist selected locale to Local Storage
## User Story:

As a website visitor who speaks Japanese, I want to be able to continue seeing the page content in Japanese on a page refresh

## Tech details

The default locale is English, so currently a page refresh means that the page reverts to English, even if the user selected Japanese.

We can fix this by saving the user's preferred locale to the browser's Local Storage

Feel free to reuse some of the logic from here:

https://github.com/WomenInSoftwareEngineeringJP/speak-her-db/blob/main/web/src/components/LocaleSwitcher.vue

(This is a Vue project and javascript, but it should give an idea on how to do it)

### PR DESCRIPTION
**Pull request by RossellaFer**
Resolves #24 

## What changed 🧐
Save selected language to local storage when the user interacts with toggle in drawer. With useEffect, at page load evaluate if the localStorage has a language value and render the content in the correct language

## How did you test it? 🧪

Added `LocaleToggle.test.tsx` test file that ensures toggle is rendered and that checks that the language is correctly switched

### CODE REVIEW

File: src/components/LocaleToggle/LocaleToggle.tsx [hunk 1]
```diff
@@ -8,24 +8,37 @@ const LocaleToggle: FC = () => {
...
+
+    useEffect(() => {
+        const savedLocale = localStorage.getItem('locale');
+        if (savedLocale) {
+            changeLanguage(savedLocale as Locale).catch((e: Error) => {
+                console.error(e)
+            })
         }
-        changeLanguage().catch((e: Error) => {
+    }, []);
```
Comment (ann-kilzer) [inline src/components/LocaleToggle/LocaleToggle.tsx:25]:
```suggestion
    }, [changeLanguage]);
```

Need to add the dependency to the hook here

**Review by ann-kilzer — COMMENTED** [review]
Thank you for writing this, first pass looks pretty good. 

We need to fix one lint warning and there is an opportunity for better tests.

File: src/components/LocaleToggle/__test__/LocaleToggle.test.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { describe, expect, it } from 'vitest'
+import { render } from '@/tests/customRender'
+import LocaleToggle from '../LocaleToggle';
+import userEvent from '@testing-library/user-event'
+
+describe('LocaleToggle', () => {
+    it('renders correctly', () => {
+        const { getByText } = render(<LocaleToggle />);
```
Comment (ann-kilzer) [inline src/components/LocaleToggle/__test__/LocaleToggle.test.tsx:8]: Let's use the [testing-library](https://testing-library.com/docs/react-testing-library/cheatsheet) style matchers for a better simulation of what the user experiences 

```suggestion
        const english = await screen.findByText('English')
        expect(english).toBeVisible()
```

You'll need to add `import { screen } from '@testing-library/react'` at the top

File: src/components/LocaleToggle/LocaleToggle.tsx (hunk 1 above)
Comment (RossellaFer) [reply src/components/LocaleToggle/LocaleToggle.tsx]: Thanks, I had to add a useCallback hook to wrap the changeLanguage function too

File: src/components/LocaleToggle/__test__/LocaleToggle.test.tsx (hunk 2 above)
Comment (RossellaFer) [reply src/components/LocaleToggle/__test__/LocaleToggle.test.tsx]: Thank you! I have made the changes to use screen in both tests

**Review by ann-kilzer — APPROVED** [review]
[LGTMeow](https://lgtmeow.com)

**Pull request merged**